#! /bin/sh /usr/share/dpatch/dpatch-run
## disable_unavailable_cpu_optimizations_r133647.dpatch by Mirco Bauer <meebey@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch taken from upstream's SVN, revision: r133647
## DP: 2009-05-06  Zoltan Varga  <vargaz@gmail.com>
## DP: * driver.c (mini_regression): Disable optimizations not supported by
## DP:   the cpu. Fixes #500019.

@DPATCH@
diff -urNad mono-2.4+dfsg~/mono/mini/driver.c mono-2.4+dfsg/mono/mini/driver.c
--- mono-2.4+dfsg~/mono/mini/driver.c	2009-02-14 00:34:26.000000000 +0100
+++ mono-2.4+dfsg/mono/mini/driver.c	2009-06-13 14:25:52.000000000 +0200
@@ -341,6 +341,9 @@
 	TestMethod func;
 	GTimer *timer = g_timer_new ();
 	MonoDomain *domain = mono_domain_get ();
+	guint32 exclude = 0;
+
+	mono_arch_cpu_optimizazions (&exclude);
 
 	if (mini_stats_fd) {
 		fprintf (mini_stats_fd, "$stattitle = \'Mono Benchmark Results (various optimizations)\';\n");
@@ -381,7 +384,7 @@
 	for (opt = 0; opt < G_N_ELEMENTS (opt_sets); ++opt) {
 		double elapsed, comp_time, start_time;
 
-		opt_flags = opt_sets [opt];
+		opt_flags = opt_sets [opt] & ~exclude;
 		mono_set_defaults (verbose, opt_flags);
 		n = opt_descr (opt_flags);
 		g_print ("Test run: image=%s, opts=%s\n", mono_image_get_filename (image), n);
